src/plans/nonlinear_least_squares_plan.jl

```diff
@@ -70,6 +70,21 @@ function get_cost(
     return 1//2 * norm(residual_values)^2
 end
 
+function get_gradient_from_Jacobian!(
```

Comment:
A short doc string would maybe be nice? And then added to the nonlinear plan docs as an internal function
Besides that, this should stay internal (not exported), since as we discussed, we might look for a more general approach anyways.

Reply:
I've added a short docstring. Could you add it to the `.md` file where you think it fits?

Reply:
That would have to wait until tomorrow if I shall do it, left my laptop in my office;
I was hoping for the objective.md file maybe as an internal subsection at the end of the Gradient objectives?